app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java

```diff
@@ -90,18 +97,35 @@ public void onClick(View view) {
         });
 
         final AutoCompleteTextView categoryInput = (AutoCompleteTextView) findViewById(R.id.categoryInput);
-        categoryInput.setAdapter(suggest(this));
+        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this, android.R.layout.simple_dropdown_item_1line, new String[]{});
+        categoryInput.setAdapter(adapter);
+        categoryInput.setThreshold(1);
+
+        categoryRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                HashMap<String, String> categoryHashMap = (HashMap<String, String>) dataSnapshot.getValue();
+                String[] categoryValues = categoryHashMap.values().toArray(new String[categoryHashMap.size()]);
```

Comment:
@BboyStatix sure the thing inside the `toArray()` shouldnt just be categoryHashMap.size()?